api-module/src/main/java/org/devridge/api/domain/emailverification/service/EmailVerificationService.java

```diff
@@ -0,0 +1,89 @@
+import org.devridge.api.exception.email.EmailVerificationInvalidException;
+import org.devridge.api.util.RandomGeneratorUtil;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.server.ResponseStatusException;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import java.time.LocalDateTime;
+
+@Service
+@RequiredArgsConstructor
+public class EmailVerificationService {
+
+    private final EmailVerificationRepository emailVerificationRepository;
+    private final EmailMessageSender emailMessageSender;
+    private final TemplateEngine templateEngine;
+    private final RandomGeneratorUtil randomGeneratorUtil;
+
+    @Value("${devridge.email.expire-minutes.signup}")
+    private int EMAIL_EXP_MINUTES;
+
+    public void sendVerificationEmail(SendEmailRequest emailVerificationRequest) {
+        EmailVerification emailVerification = createEmailVerification(emailVerificationRequest.getEmail(), EMAIL_EXP_MINUTES);
+        emailVerificationRepository.save(emailVerification);
+
+        Context context = new Context();
+        context.setVariable("verificationCode", emailVerification.getContent());
+
+        String htmlContent = templateEngine.process("verificationEmailTemplate", context);
+
+        emailMessageSender.sendSimpleMessage(
+                emailVerification.getReceiptEmail(),
+                "[devridge] 가입 인증번호 안내",
+                htmlContent
+        );
+    }
+
+    private EmailVerification createEmailVerification(String email, int expMinutes) {
+        String content = String.valueOf(randomGeneratorUtil.generateFourDigitNumber());
+        LocalDateTime expiredAt = LocalDateTime.now().plusMinutes(expMinutes);
+
+        return EmailVerification.builder()
+                .receiptEmail(email)
+                .content(content)
+                .checkStatus(false)
+                .expireAt(expiredAt)
+                .build();
+    }
+
+    @Transactional
+    public void checkVerificationCode(String email, String code) {
+        EmailVerification emailVerification = emailVerificationRepository.findTopByReceiptEmailOrderByCreatedAtDesc(email)
+                .orElseThrow(() -> new EmailVerificationInvalidException());
+        System.out.println("emailVerification = " + emailVerification);
+        
+        LocalDateTime current = LocalDateTime.now();
+
+        validateEmailVerification(emailVerification, current);
+        validateVerificationCode(code, emailVerification.getContent());
+
+        emailVerification.changeCheckStatus();
+
+        emailVerificationRepository.save(emailVerification);
+    }
+
+    private static void validateEmailVerification(EmailVerification emailVerification, LocalDateTime current) {
+        if (emailVerification.isCheckStatus() == true) {
```

Comment:
중요하진 않은데 아래처럼 하시면 가독성에 더 도움이 될 것 같슴당
```suggestion
        if (emailVerification.isCheckStatus()) {
```